### ISSUE TITLE
Why is dash changed to underscore?

### ISSUE THREAD

**Issue by justin-calleja** [HashNuke/asdf-nodejs#2]
After doing: `npm install -g private-bower`

`private-bower` is not available on the PATH. In .asdf/shims/private_bower, I can see it's doing:

`exec /Users/justin/.asdf/bin/private/asdf-exec nodejs .npm/bin/private_bower "$@"`

But `private_bower`, although available on the PATH, gives me "No such command in 0.12.2 of nodejs".

That's because in .asdf/installs/nodejs/0.12.2/.npm/bin/, I have `private-bower` not `private_bower`...

To fix this I have to rename .asdf/shims/private_bower to .asdf/shims/private-bower and change it's content to use the dash:

`exec /Users/justin/.asdf/bin/private/asdf-exec nodejs .npm/bin/private-bower "$@"`

So, why are dashes changed to underscore when installing npm packages?

**Comment by HashNuke** [comment]
I'll check this out. It asdf shouldn't touch the executable file names.

**Comment by justin-calleja** [comment]
Awesome :+1:

**Comment by HashNuke** [comment]
@justin-calleja Thank you for reporting this. npm install hook was getting the name of the binary from the wrong place.

The npm install hooks will have to be fixed in every nodejs version installed. 

**To make it easy, here's a small script you can copy paste to your terminal:**

```
asdf list nodejs | while read nodejs_version; do
  echo "Fixing $nodejs_version"
  curl -s https://raw.githubusercontent.com/HashNuke/asdf-nodejs/master/npm-hooks/postinstall -o installs/nodejs/${nodejs_version}/.npm/lib/node_modules/.hooks/postinstall
done
```

Additionally I've created an issue https://github.com/HashNuke/asdf-nodejs/issues/3 to centralize install hooks. That shoud make fixing or updating install hooks easier

**Comment by justin-calleja** [comment]
Many thanks @HashNuke !!

btw, before reading your last comment, I tried pulling the change in the plugin and that didn't work - seems like that only changes .asdf/plugins and not .asdf/installs

I tried `asdf plugin-update nodejs` - that didn't work either. Should that have worked?

In any case, both your script and removing the plugin and re-installing node versions put your changes in effect. Works like a charm now :smile:

**Comment by HashNuke** [comment]
Yes that is why I gave you the script. So that you won't have to re-install
nodejs versions :)

**Comment by justin-calleja** [comment]
haha, no worries - I like breaking things :smile: 

Cheers though, might come in handy next time (and learnt about `https://raw.githubusercontent.com/` from it :+1:)
